src/bindgen/ir/structure.rs

```diff
@@ -24,6 +24,7 @@ pub struct Struct {
     pub generic_params: Vec<String>,
     pub documentation: Documentation,
     pub functions: Vec<Function>,
+    pub destructor: Option<Function>,
```

Comment:
Seems like my c++ got a bit rusty :smile:  
What about the following idea:
If there is a destructor the exported struct is likely no POD and does not implement `Copy` on rust side. Therefore the copy constructor and copy assignment operator should be disabled. ( A future version may allow them by defining a clone function(like the destructor) that is internally called)
For me this seems like the solution that matches the underlying behavior of the rust code.

> I think whatever the solution is, it needs to be complete so we can't just generate destructor.

I've pushed the current implementation to see what others think about the idea. I've developed this while trying to generate a usable api for our internal project. Internally all my functions to free some allocated memory are implemented to reset the struct in a non allocating state, meaning free all memory and then set all pointers to null pointers. So the copy constructor should not harm this implementation. (But this is defined by my implementation, not by the generated header, so this is a not so great solution)